lib/util/Map.v3

```diff
@@ -110,6 +107,39 @@ class HashMap<K, V> extends PartialMap<K, V> {
 			}
 		}
 	}
+	// Remove {key} from this map. Return true if {key} existed.
+	def remove(key: K) -> bool {
+		var c = cache;
+		var found = false;
+		// if cache hit, clear cache
+		if (c != null && (c.key == key || equals(c.key, key))) {
+			found = true;
+			cache = null;
+		}
+		// search table
+		if (table != null) {
+			for (i < table.length) {
```

Comment:
Sorry I didn't notice this before. This should do a hashed search; no need to search the entire table, only the bucket chain in the right hash slot.